lldb/source/Plugins/Language/Swift/SwiftOptional.cpp

```diff
@@ -44,24 +44,46 @@ static PointerOrSP
 ExtractSomeIfAny(ValueObject *optional,
                  bool synthetic_value = false) {
   if (!optional)
-    return nullptr;
+    return PointerOrSP("Can't get value from null optional");
 
   static ConstString g_Some("some");
   static ConstString g_None("none");
+  
+  const char *name = optional->GetName().GetCString();
+  if (!name)
+    name = "<unknown>";
 
   ValueObjectSP non_synth_valobj = optional->GetNonSyntheticValue();
-  if (!non_synth_valobj)
-    return nullptr;
+  if (!non_synth_valobj) {
+    std::string err_msg("Can't get the underlying value object for '");
+    err_msg.append(optional->GetName().GetCString());
+    err_msg.append("'");
+    return PointerOrSP(err_msg);
+  }
 
   ConstString value(non_synth_valobj->GetValueAsCString());
-
-  if (!value || value == g_None)
+  
+  if (!value) {
+    // FIXME: I can't see a way to get an error message from GetValueAsCString
+    std::string err_msg("Can't get the underlying value object's value '");
+    err_msg.append(name);
+    err_msg.append("'");
+    return PointerOrSP(err_msg);
+  }
+  
+  if (value == g_None)
     return nullptr;
 
-  PointerOrSP value_sp(
-      non_synth_valobj->GetChildMemberWithName(g_Some, true).get());
-  if (!value_sp)
-    return nullptr;
+  ValueObjectSP valobj_sp = non_synth_valobj->GetChildMemberWithName(g_Some, 
+                                                                     true);
+  if (!valobj_sp) {
+    std::string err_msg("Failed to get the 'some' field from optional '");
```

Comment:
I could see this message being too low level (most won't know what a a some field is in an optional), but I also don't have a better suggestion…

Reply:
These are mostly for us and to convey that "something" went wrong.  We really don't know enough here to tell the user usefully want actually went wrong.